src/Tribe/Views/V2/Widgets/Widget_QR_Code.php

```diff
@@ -254,10 +260,17 @@ public function setup_admin_fields() {
 		];
 
 		$events = tribe_get_events( $args );
-		foreach ( $events as $event ) {
+		if ( ! empty( $events ) ) {
+			foreach ( $events as $event ) {
+				$event_options[] = [
+					'value' => $event->ID,
+					'text'  => $event->ID . ' - ' . $event->post_title,
+				];
+			}
+		} else {
 			$event_options[] = [
-				'value' => $event->ID,
-				'text'  => $event->ID . ' - ' . $event->post_title,
+				'value' => '',
+				'text'  => _x( 'There are no Events created yet.', 'No events created yet', 'the-events-calendar' ),
```

Comment:
Same as above, I don't think context is necessary here, plus a minor textual suggestion.

```suggestion
				'text'  => __( 'No Events have been created yet.', 'the-events-calendar' ),
```